docs/docs/20-usage/90-advanced-usage.md

```diff
@@ -127,3 +127,40 @@ WOODPECKER_ENVIRONMENT=first_var:value1,second_var:value2
 ```
 
 Note that this tightly couples the server and app configurations (where the app is a completely separate application). But this is a good option for truly global variables which should apply to all steps in all pipelines for all apps.
+
+## Docker in docker (dind) setup
+
+:::warning
+This set up will only work on trusted repositories and for security reasons should only be used in private environments. See [project settings](./75-project-settings.md#trusted) to enable trusted mode.
+:::
+
+The snippet below shows how a step can communicate with the docker deamon via a `docker:dind` service.
+
+The most important bit ist to allow the dind service to write/read certificates on a mounted volume that is also shared with the step that will contact the daemon (it needs the service's CA to do so successfully).
+
+The `DOCKER_*` environment variables in the step are generic settings that should work with your framework of choice (e.g. TestContainers or similar):
```

Comment:
What does "TestContainers or similar" mean here exactly? Can't make much sense out of that part.

Reply:
Thanks for the incredibly quick reply!

My usecase is not to build images, but to use a framework like [testcontainers](https://testcontainers.com/) to start/stop containers during test, so that close-to-production infrastructure can be started on demand. One could compare it to the services feature of woodpecker but allows more advanced scenarios.

I'd be happy to remove that comment if it is a bit too specific / out of scope.

I will gladly add that for normal oci building scenarios, the buildx plugin is the way to go

Reply:
Ah, didn't know that project before. Makes sense then 👍️ I think when you add a link, it is fine and even a great example of when to use this approach. Bundled with the note that buildx should be used for building, this looks like a great addition then.

Reply:
Thanks for the feedback. I added two examples of frameworks in the end, with a couple of links.

If there's anything I should add/modify, please let me know!